hrabsence/CRM/HRAbsence/BAO/HRAbsenceType.php

```diff
@@ -154,12 +154,32 @@ public static function getActivityStatus($return = 'label') {
     }
     return $result;
   }
-
-  public static function del($absenceTypeId) {
+  
+  /**
+   * Delete Absence Type
+   * 
+   * @param int $absenceTypeId
+   *   ID of Absence Type to be deleted
+   * @throws CRM_Core_Exception
+   *   If Absence Type is Being used on existin leave requests
+   */
+  public static function del($absenceTypeId) {    
+    // Load Absence Type
     $absenceType = new CRM_HRAbsence_DAO_HRAbsenceType();
     $absenceType->id = $absenceTypeId;
     $absenceType->find(TRUE);
 
+    // Validate if Absence Type is used in existing requests
```

Comment:
here, instead of a comment you can make this a lot more readable by extracting this logic to a method and then:
```php
if(self::thereAreLeaveRequestsForAbsenceType($absenceType)) {
  throw new CRM_Core_Exception(ts("You can not delete this absence type -- it is assigned to %1 existing leave request(s). If you do not want this absence type to be used going forward, consider disabling it instead.", array(1 => ($creditCount + $debitCount))))
}
```

With that, you wouldn't be able to include the number of leave requests on the exception message though. If you really need to do that you can use the countLeaveRequests instead, which should return all the leave requests for an Absence Type, and not for an activity type (see my other comment on that method)